File: fixtures/java/type_inference/negated_checks/negated_checks.java
public class NegatedChecks {

    public static String handleNotInstance(Object obj) {
        if (!(obj instanceof String)) {
            return "not a string";
        }
        return ((String) obj).toUpperCase();
    }

    public static String handleNull(String value) {
        if (value != null) {
            return value;
        }
        return "missing";
    }

    public static void main(String[] args) {
        System.out.println(handleNotInstance(42));
        System.out.println(handleNull("hello"));
    }
}

File: fixtures/typescript/type_inference/negated_checks/negated_checks.ts
function handleNotNull(value: string | null): string {
    if (value !== null) {
        return value.toUpperCase();
    }
    return "missing";
}

function handleNegated(value: string | number): string {
    if (typeof value !== "string") {
        return value.toFixed(1);
    }
    return value;
}

function handleFalsy(value: string | undefined): string {
    if (!value) {
        return "empty";
    }
    return value;
}

const a = handleNotNull("hello");
const b = handleNegated(42);
const c = handleFalsy(undefined);

export { handleNotNull, handleNegated, handleFalsy, a, b, c };
